components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,317 @@
+        securityEventTokenPayload.setIss(getURL());
+        securityEventTokenPayload.setIat(System.currentTimeMillis());
+        securityEventTokenPayload.setJti(UUID.randomUUID().toString());
+        securityEventTokenPayload.setRci(getCorrelationID());
+        Map<String, EventPayload> eventMap = new HashMap<>();
+        eventMap.put(eventUri, eventPayload);
+        securityEventTokenPayload.setEvent(eventMap);
+        return securityEventTokenPayload;
+    }
+
+    /**
+     * Get correlation id from the MDC.
+     * If not then generate a random UUID, add it to MDC and return the UUID.
+     *
+     * @return Correlation id
+     */
+    public String getCorrelationID() {
+
+        String correlationID = MDC.get(CORRELATION_ID_MDC);
+        if (StringUtils.isBlank(correlationID)) {
+            correlationID = UUID.randomUUID().toString();
+            MDC.put(CORRELATION_ID_MDC, correlationID);
+        }
+        return correlationID;
+    }
+
+    private void setLocalUserClaimsToAuthenticatedUser(AuthenticatedUser authenticatedUser,
+                                                       AuthenticationContext context) {
+
+        Map<String, String> claimMappings = (Map<String, String>) context.getParameters()
+                .get(Constants.SP_TO_CARBON_CLAIM_MAPPING);
+
+        if (claimMappings == null) {
+            log.debug("No local claim mappings found for the authenticated user from the context.");
+            return;
+        }
+
+        Map<ClaimMapping, String> userAttributes = authenticatedUser.getUserAttributes();
+
+        if (userAttributes == null) {
+            userAttributes = new HashMap<>();
+        }
+
+        for (Map.Entry<ClaimMapping, String> entry : userAttributes.entrySet()) {
+
+            ClaimMapping claimMapping = entry.getKey();
+            Claim localClaim = claimMapping.getLocalClaim();
+
+            if (claimMappings.containsKey(localClaim.getClaimUri())) {
+                localClaim.setClaimUri(claimMappings.get(localClaim.getClaimUri()));
+            }
+        }
+    }
+
+    /**
+     * Get the tenant qualified URL.
+     *
+     * @return Tenant qualified URL.
+     */
+    public String getURL() {
+
+        return getURL(null);
+    }
+
+    /**
+     * Get the tenant qualified URL with path.
+     *
+     * @param endpoint Endpoint.
+     * @return Tenant qualified URL.
+     */
+    public String getURL(String endpoint) {
+
+        try {
+            ServiceURLBuilder builder = ServiceURLBuilder.create();
+            if (endpoint != null && !endpoint.isEmpty()) {
+                builder.addPath(endpoint);
+            }
+            return builder.build().getAbsolutePublicURL();
+        } catch (URLBuilderException e) {
+            log.debug("Error occurred while building the endpoint URL with tenant qualified URL.", e);
+        }
+        return endpoint != null ? endpoint : "";
+    }
```

Comment:
It looks like this is not used other than line 269. In such case, do we need this function. Why not just use this within the function in line 269